ImguiMapEditor/Controllers/SearchController.cpp

```
@@ -75,8 +77,94 @@ void SearchController::onMapLoaded(
 
+    // Wire async text search
+    search_results_widget_->setSearchAsyncCallback(
+        [this](const std::string& query, bool search_items, bool search_creatures) {
+            searchTextAsync(query, search_items, search_creatures);
+        });
+
     current_client_data_ = client_data;
 }
 
+void SearchController::searchUniqueAsync() {
+    if (!map_search_service_) return;
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)